TEveEventDisplay/src/TEveMu2e_base_classes/TEveMu2eCustomHelix.h

```diff
@@ -3,27 +3,22 @@
 
 #include <TObject.h>
 #include <THelix.h>
-#include <TPolyLine3D.h>
+#include <TEveLine.h>
 #include "RecoDataProducts/inc/HelixSeed.hh"
 #include "RecoDataProducts/inc/KalSeed.hh"
+#include "RecoDataProducts/inc/TrkExtTraj.hh"
 
-#include <TEveTrackPropagator.h>
+using namespace mu2e;
 
 namespace mu2e {
   class   TEveMu2eCustomHelix: public TEveLine {
     public:
-       #ifndef __CINT__
-      explicit  TEveMu2eCustomHelix(){};
-      TEveMu2eCustomHelix(const TEveMu2eCustomHelix &helix) { fKalSeed = helix.fKalSeed;} ;
-      TEveMu2eCustomHelix(HelixSeed hseed){fHelixSeed = hseed;};
-      TEveMu2eCustomHelix(KalSeed kseed){
-        fKalSeed = kseed;
-        this->Momentum = fKalSeed.helix()->helix().momentum();
-        this->PDGcode = fKalSeed.particle().particleType();
-        this->Charge = fKalSeed.particle().charge();
-        this->Mass = fKalSeed.particle().mass();
-      };
-      virtual ~ TEveMu2eCustomHelix(){};
+      #ifndef __CINT__
+      explicit  TEveMu2eCustomHelix();
```

Comment:
There are two issues with the 4 constructors.

First the constructors from HelixSeed and KalSeed should have their arguments passed by const & - the present code makes a copy.  Makes a copy of the copy and then deletes the first copy.

The other issue is the use of explicit.  It s only meaningful for constructors with exactly one argument.  I disables implicit type conversions.  It is ignored in all other cases.

As written, if you were to provide a HelixSeed as an actual argument when the function prototype expects a TEveMu2eCustomHelix, then the complier would create a temporary TEveCustomHelix and pass that as the argument.  If you want to disable this behaviour, add explicit in the c'tor that takes a HelixSeed; if you do so, it will produce a compiler diagnostic which is almost certainly what you want.  Please take the explicit off of the default c'tor and add it to all single argument c'tors.